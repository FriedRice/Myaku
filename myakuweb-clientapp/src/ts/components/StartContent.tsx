/** @module Start page content for MyakuWeb */

import GettingStartedTile from './GettingStartedTile';
import React from 'react';
import { Search } from '../types';
import WhatIsMyakuTile from './WhatIsMyakuTile';

interface StartContentProps {
    onSearchSubmit: (search: Search) => void;
}
type Props = StartContentProps;

const StartContent: React.FC<Props> = function(props) {
    return (
        <div className='start-tile-container'>
            <WhatIsMyakuTile />
            <GettingStartedTile onSearchSubmit={props.onSearchSubmit} />
        </div>
    );
};

export default StartContent;
