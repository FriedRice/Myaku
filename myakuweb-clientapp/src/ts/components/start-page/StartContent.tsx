/** @module Start page content for MyakuWeb */

import GettingStartedTile from 'ts/components/start-page/GettingStartedTile';
import React from 'react';
import { Search } from 'ts/types/types';
import WhatIsMyakuTile from 'ts/components/start-page/WhatIsMyakuTile';

interface StartContentProps {
    onSearchSubmit: (search: Search) => void;
}
type Props = StartContentProps;

const StartContent: React.FC<Props> = function(props) {
    return (
        <div className='start-tile-container'>
            <WhatIsMyakuTile />
            <GettingStartedTile onSearchSubmit={props.onSearchSubmit} />
        </div>
    );
};

export default StartContent;
