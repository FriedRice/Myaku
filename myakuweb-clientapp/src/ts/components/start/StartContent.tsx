/**
 * Start page content for MyakuWeb.
 * @module ts/components/start/StartContent
 */

import GettingStartedTile from 'ts/components/start/GettingStartedTile';
import React from 'react';
import WhatIsMyakuTile from 'ts/components/start/WhatIsMyakuTile';
import { useEffect } from 'react';


const StartContent: React.FC<{}> = function() {
    useEffect(function() {
        document.title = 'Myaku';
    }, []);

    return (
        <div className='start-tile-container'>
            <WhatIsMyakuTile />
            <GettingStartedTile />
        </div>
    );
};

export default StartContent;
