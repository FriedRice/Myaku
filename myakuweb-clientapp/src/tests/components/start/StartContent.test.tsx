/**
 * Tests for the StartContent component.
 * @module tests/components/start/StartContent.test
 */

import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import StartContent from 'ts/components/start/StartContent';

import {
    mount,
    shallow,
} from 'enzyme';


describe('<StartContent />', function() {
    it('renders correctly', function() {
        const wrapper = shallow(<StartContent />);
        expect(wrapper).toMatchSnapshot();
    });

    it('sets document title', function() {
        mount(
            <MemoryRouter>
                <StartContent />
            </MemoryRouter>
        );
        expect(document.title).toBe('Myaku');
    });
});
