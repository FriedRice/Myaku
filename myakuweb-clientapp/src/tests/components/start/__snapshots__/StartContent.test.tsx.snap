// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartContent /> renders correctly 1`] = `
<div
  className="start-tile-container"
>
  <WhatIsMyakuTile />
  <GettingStartedTile />
</div>
`;
